src/contrib/mlir/compiler.cpp

```diff
@@ -301,131 +303,136 @@ namespace ngraph
             {
-                return compiler.create_binary_op<mlir::NGGreaterOp>(ng_node);
+                return compiler.create_generic_op<mlir::NGGreaterOp>(ng_node);
             }
 
             template <>
             mlir::Value* MLIRCompiler::COMPILE_OP_DECL(ngraph::op::Less)
             {
-                return compiler.create_binary_op<mlir::NGLessOp>(ng_node);
+                return compiler.create_generic_op<mlir::NGLessOp>(ng_node);
             }
 
             template <>
             mlir::Value* MLIRCompiler::COMPILE_OP_DECL(ngraph::op::Maximum)
             {
-                return compiler.create_binary_op<mlir::NGMaxOp>(ng_node);
+                return compiler.create_generic_op<mlir::NGMaxOp>(ng_node);
             }
 
             template <>
             mlir::Value* MLIRCompiler::COMPILE_OP_DECL(ngraph::op::Minimum)
             {
-                return compiler.create_binary_op<mlir::NGMinOp>(ng_node);
+                return compiler.create_generic_op<mlir::NGMinOp>(ng_node);
             }
 
-            template <>
-            mlir::Value* MLIRCompiler::COMPILE_OP_DECL(ngraph::op::ArgMax)
+           template <>
+            mlir::Value* MLIRCompiler::COMPILE_OP_DECL(ngraph::op::Dot)
             {
-                return compiler.create_index_reduction<mlir::NGArgMaxRedOp>(ng_node);
+                return compiler.create_generic_op<mlir::NGDotOp>(ng_node);
             }
 
             template <>
-            mlir::Value* MLIRCompiler::COMPILE_OP_DECL(ngraph::op::ArgMin)
+            mlir::Value* MLIRCompiler::COMPILE_OP_DECL(ngraph::op::ArgMax)
             {
-                return compiler.create_index_reduction<mlir::NGArgMinRedOp>(ng_node);
+                auto* idx_red = static_cast<const ngraph::op::util::IndexReduction*>(ng_node);
+                mlir::Value* result = compiler.create_generic_op<mlir::NGArgMaxRedOp>(ng_node);
+                mlir::Operation* op = result->getDefiningOp();
+                mlir::ArrayAttr red_axes_attr = compiler.m_builder->getI64ArrayAttr({(int64_t)idx_red->get_reduction_axis()});
+                op->setAttr("axes", red_axes_attr);
+                return result;
             }
 
             template <>
-            mlir::Value* MLIRCompiler::COMPILE_OP_DECL(ngraph::op::Dot)
+            mlir::Value* MLIRCompiler::COMPILE_OP_DECL(ngraph::op::ArgMin)
             {
-                return compiler.create_binary_op<mlir::NGDotOp>(ng_node);
+                auto* idx_red = static_cast<const ngraph::op::util::IndexReduction*>(ng_node);
+                mlir::Value* result = compiler.create_generic_op<mlir::NGArgMinRedOp>(ng_node);
+                mlir::Operation* op = result->getDefiningOp();
+                mlir::ArrayAttr red_axes_attr = compiler.m_builder->getI64ArrayAttr({(int64_t)idx_red->get_reduction_axis()});
+                op->setAttr("axes", red_axes_attr);
+                return result;
```

Comment:
I would suggest keeping `create_index_reduction` and make it call `create_generic_op` to avoid 360-365 and 371-376.